bigquery/v2/query/helper.go

```diff
@@ -0,0 +1,101 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package query
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"cloud.google.com/go/bigquery/v2/apiv2/bigquerypb"
+	"cloud.google.com/go/bigquery/v2/apiv2_client"
+	"cloud.google.com/go/internal/uid"
+	"github.com/googleapis/gax-go/v2"
+	"google.golang.org/api/option"
+)
+
+// Helper for running queries in BigQuery. It is a lightweight wrapper
+// around the auto-generated BigQuery v2 client, focused on query operations.
+type Helper struct {
+	c         *apiv2_client.Client
+	projectID string
+}
+
+// NewHelper creates a new query helper. This helper should be reused instead of
+// created per-request.
+func NewHelper(c *apiv2_client.Client, projectID string, opts ...option.ClientOption) (*Helper, error) {
```

Comment:
something for followup in subsequent PRs:  should it accept ClientOption if it's not instantiating clients?